validator/client/src/main/java/tech/pegasys/teku/validator/client/duties/synccommittee/SyncCommitteeScheduledDuties.java

```diff
@@ -77,6 +77,9 @@ public boolean requiresRecalculation(final Bytes32 newHeadDependentRoot) {
   public SafeFuture<DutyResult> performProductionDuty(final UInt64 slot) {
     lastSignatureBlockRoot = chainHeadTracker.getCurrentChainHead(slot);
     lastSignatureSlot = Optional.of(slot);
+    if (assignments.isEmpty() && lastSignatureBlockRoot.isEmpty()) {
+      return SafeFuture.completedFuture(DutyResult.NO_OP);
+    }
```

Comment:
We're still performing this check after we get the current chain head but isn't the point to make sure we don't ask for the chain head if we aren't going to use it? There's no point going beyond here even if the chain head is known.